### PR TITLE
Feat/datagrid step 4

### PR DESCRIPTION
**Pull request by aesteves60**
Add custom formatter by column

### CODE REVIEW

File: packages/components/datagrid/src/components/osds-datagrid/core/controller.ts [hunk 1]
```diff
@@ -17,8 +17,9 @@ class OdsDatagridController {
             field: column.field,
             headerSort: column.isSortable ?? false,
             headerHozAlign: 'center',
+            vertAlign: 'middle',
             titleFormatter: (cell: CellComponent) => this.getOdsText(cell.getValue(), ODS_TEXT_SIZE._500),
-            formatter: (cell: CellComponent) => this.getOdsText(cell.getValue(), ODS_TEXT_SIZE._400),
+            formatter: (cell: CellComponent) => column.formatter?.(cell.getValue()) ?? this.getOdsText(cell.getValue(), ODS_TEXT_SIZE._400),
```
Comment (dpellier) [inline packages/components/datagrid/src/components/osds-datagrid/core/controller.ts:22]: would be nice to add a dedicated sandbox page with a very large datagrid (10000+ rows) to ensure we don't have any perf issue

**Review by dpellier — CHANGES_REQUESTED** [review]

**Review by dpellier — CHANGES_REQUESTED** [review]
Also add a `build:ci` command on datagrid package otherwise doc will not be build

**Review by dpellier — CHANGES_REQUESTED** [review]
Cursor is always `pointer` on each row, should be default one until row get interactive (like draggable)

**Review by feoche — CHANGES_REQUESTED** [review]
UI review : 
- [x] In "Default" Storybook story, `rowHeight` property should be 52px (3.25rem)
- [x] Changing story parameters does not trigger component re-render
- [x] Cursor should be `auto` when hovering rows
- [ ] [Safari] column with icon is 1px off at the bottom of each cell (can be fixed with --webkit-inline-flex on row)
- [x] Sortable column label height is different to non-sortable label
- [x] Header row has different height than the `rowHeight` (to be discussed)

- [x] (Switching stories between "Default" and "Large Data" breaks its display - hard to reproduce)
- [x] Remove control on Large Data Page

**Review by Leotheluck — APPROVED** [review]

**Review by astagnol — CHANGES_REQUESTED** [review]
Click on isSelectable control, then click again. Header is broken:
Capture d’écran 2023-10-24 à 17 40 54

**Comment by feoche** [comment]
- [ ] Removing all columns does not get the "noResultLabel" to appear
- [ ] Header is always as "fixed", without a drop shadow (to discuss with Design)

**Review by astagnol — APPROVED** [review]

**Review by feoche — APPROVED** [review]

**Pull request merged**